packages/flutter_tools/lib/src/macos/xcode.dart

```diff
@@ -29,6 +29,10 @@ const int kXcodeRequiredVersionMajor = 11;
 const int kXcodeRequiredVersionMinor = 0;
 const int kXcodeRequiredVersionPatch = 0;
 
+const int kXcodeRecommendedVersionMajor = 12;
```

Comment:
Can we replace these with our existing Version class?  https://github.com/flutter/flutter/blob/e605b7c2250e803f46017b1d544fc8c297007546/packages/flutter_tools/lib/src/base/version.dart#L8

Then you can just `if (currentVersion >= recommendedVersion) { ... }`

Reply:
Good idea, done.